met-web/src/components/engagement/view/EmailModal.tsx

```diff
@@ -48,10 +48,14 @@ const EmailModal = ({ defaultPanel, open, handleClose }: EmailModalProps) => {
                 type: EmailVerificationType.Survey,
                 language: language.id,
             });
-            window.snowplow('trackSelfDescribingEvent', {
-                schema: 'iglu:ca.bc.gov.met/verify-email/jsonschema/1-0-0',
-                data: { survey_id: savedEngagement.surveys[0].id, engagement_id: savedEngagement.id },
-            });
+            try {
+                window.snowplow('trackSelfDescribingEvent', {
+                    schema: 'iglu:ca.bc.gov.met/verify-email/jsonschema/1-0-0',
+                    data: { survey_id: savedEngagement.surveys[0].id, engagement_id: savedEngagement.id },
+                });
+            } catch (error) {
+                console.log(error);
```

Comment:
Some more context for the error would be nice here. It may be hard to track this down in the code if it throws without a message.

```suggestion
                console.log('Snowplow error:', error);
```

Reply:
Added, thanks.